Quadriga/src/main/java/edu/asu/spring/quadriga/web/conceptcollection/ConceptcollectionController.java

```diff
@@ -161,18 +165,24 @@ private void fillModel(String collectionId, ModelMap model, String username) thr
     @RequestMapping(value = "auth/conceptcollections/{collection_id}/searchitems", method = RequestMethod.GET)
     public String conceptSearchHandler(@PathVariable("collection_id") String collection_id, HttpServletRequest req,
             ModelMap model) throws QuadrigaStorageException, QuadrigaAccessException {
+        try {
+            ConceptpowerReply conReply = conceptControllerManager.search(req.getParameter("name"),
+                    req.getParameter("pos"));
 
-        ConceptpowerReply conReply = conceptControllerManager.search(req.getParameter("name"), req.getParameter("pos"));
-        if (conReply != null) {
-            List<ConceptEntry> lists = conReply.getConceptEntry();
-            lists.sort(new Comparator<ConceptEntry>() {
+            if (conReply != null) {
+                List<ConceptEntry> lists = conReply.getConceptEntry();
+                lists.sort(new Comparator<ConceptEntry>() {
 
-                @Override
-                public int compare(ConceptEntry o1, ConceptEntry o2) {
-                    return o1.getLemma().toLowerCase().compareTo(o2.getLemma().toLowerCase());
-                }
-            });
-            model.addAttribute("result", conReply.getConceptEntry());
+                    @Override
+                    public int compare(ConceptEntry o1, ConceptEntry o2) {
+                        return o1.getLemma().toLowerCase().compareTo(o2.getLemma().toLowerCase());
+                    }
+                });
+                model.addAttribute("result", conReply.getConceptEntry());
+            }
+        } catch (HttpMessageNotReadableException hex) {
+            logger.error("error:", hex);
+            return "auth/storageissue";
```

Comment:
if you just throw a Quadriga exception the general exception handler should take care of displaying it. this makes sense if you actually do something in response to it.